src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/instance_trial_of_the_crusader.cpp

```diff
@@ -879,8 +875,25 @@ class instance_trial_of_the_crusader : public InstanceMapScript
                     {
                         if( Creature* c = instance->GetCreature(NPC_JaraxxusGUID) )
                         {
-                            c->Yell("Banished to the Nether!", LANG_UNIVERSAL);
-                            c->PlayDirectSound(16146, 0);
+                            uint8 textID = urand(0, 1);
+                            switch (textID)
+                            {
+                                case 0:
+                                {
+                                    c->AI()->Talk(9);
+                                    c->PlayDirectSound(16145, 0);
```

Comment:
why are we using magic numbers, and why isnt the sound tied to the text?

Reply:
> why are we using magic numbers, and why isnt the sound tied to the text?

I have no idea, that's how it was done, ask the person who wrote the script at the time, I only made some modifications to the gossip and then I started to look at some things related to the mechanics of the event.

Reply:
but you are changing this entire codeblock. If you didnt write this code, then where did you pull it from?

Reply:
> but you are changing this entire codeblock. If you didnt write this code, then where did you pull it from?

Yes, I changed the code, now I understand what you mean. The thing is that the NPC has 2 different dialogs, but in the database, it has only 1. For that reason, the original author left the code that way, and the truth is that I didn't edit it in that aspect. But I agree, that you could add the sound inside the creature_text, but for that, you would have to do a sniff, because it seems that there is an error in the text. One of them is missing.

Reply:
just don't use hardcoded numbers in the core.